src/Examples/Program.cs

```diff
@@ -77,33 +84,96 @@ internal static async Task MainAsync(string[] args)
                 throw new InvalidOperationException("Invalid input!");
 
             // cuz Singleplayer = 0
-            var gameMode = (ZPlayMode) targetGameMode - 1;
+            var gameMode = (ZPlayMode)targetGameMode - 1;
 
             #endregion
 
             switch (gameMode)
             {
+                // create and run singleplayer game
                 case ZPlayMode.Singleplayer:
-                    var gameProcess = await gameFactory.CreateSingleAsync(new ZSingleParams { Game = game });
 
-                    // track game pipe
-                    gameProcess.Pipe += (sender, pipeArgs) => Console.WriteLine(pipeArgs.FullMessage);
+                    // create game
+                    var gameProcess = await _gameFactory.CreateSingleAsync(new ZSingleParams { Game = game });
 
-                    // run game process
-                    var runResult = await gameProcess.RunAsync();
-
-                    Console.WriteLine($"Run result {runResult}");
+                    // run and track game pipe
+                    await _RunAndTrack(gameProcess);
 
                     break;
                 case ZPlayMode.Multiplayer:
+
+                    await _MultiplayerHandler(game);
+
                     break;
 
                 case ZPlayMode.CooperativeHost:
                 case ZPlayMode.CooperativeClient:
                 case ZPlayMode.TestRange:
+                    //testrange process 
+             var testRange = await _gameFactory.CreateTestRangeAsync(new ZTestRangeParams { Game = game });
+
+                  //run test range process handler 
+             await _RunAndTrack(testRange);
+
+                    break;
                 default:
                     throw new ArgumentOutOfRangeException();
             }
         }
+
+        internal static async Task _RunAndTrack(IZRunGame gameProcess)
+        {
+            var resetEvent = new ManualResetEvent(false);
+
+            // track game pipe
+            gameProcess.Pipe += (sender, pipeArgs) =>
+            {
+                Console.WriteLine(pipeArgs.FullMessage);
+
+                // return from _RunAndTrack if game closed
+                if (pipeArgs.SecondPart.Contains("Closed"))
+                {
+                    resetEvent.Set();
+                }
+            };
+
+            // run game process
+            var runResult = await gameProcess.RunAsync();
+
+            // the result will be an enumeration
+            // that will help determine if the game was launched successfully (returned directly by the ZClient)
+            if (runResult != ZRunResult.Success)
+            {
+                // TODO: Do some stuff here
+            }
+
+            resetEvent.WaitOne();
+        }
+
+        internal static async Task _MultiplayerHandler(ZGame game)
+        {
+            //table line for top and bottom 
```

Comment:
@nintend01337 Please use a space before writing a comment
For example:
//some comment
// some connect